src/figuresClasses.ts

```diff
@@ -2,10 +1,90 @@
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  constructor(
+    public color: string,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    if (a <= 0) {
+      throw new Error("Side 'a' must be greater than zero.");
+    }
+
+    if (b <= 0) {
+      throw new Error("Side 'b' must be greater than zero.");
+    }
+
+    if (c <= 0) {
+      throw new Error("Side 'c' must be greater than zero.");
+    }
+
+    const sides = [a, b, c].sort((x, y) => x - y);
+
+    if (sides[2] >= sides[0] + sides[1]) {
+      throw new Error(
+        `The longest side of a triangle must be less than the sum of the other two sides.`,
+      );
+    }
+  }
+
+  shape: string = 'triangle';
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  constructor(
+    public color: string,
+    private radius: number,
+  ) {
+    if (radius <= 0) {
```

Comment:
The `radius` parameter should be prefixed with `this.` when used inside the constructor to refer to the class property. This will fix the reference error when checking the radius value.